src/modules/local_position_estimator/sensors/landing_target.cpp

```diff
@@ -0,0 +1,121 @@
+#include "../BlockLocalPositionEstimator.hpp"
+#include <systemlib/mavlink_log.h>
+#include <matrix/math.hpp>
+
+extern orb_advert_t mavlink_log_pub;
+
+static const uint64_t 	TARGET_TIMEOUT =   2000000; // [us]
+
+void BlockLocalPositionEstimator::landingTargetInit()
+{
+	if (_target_mode.get() == Target_Moving) {
+		// target is in moving mode, do not initialize
+		return;
+	}
+
+	Vector<float, n_y_target> y;
+
+	if (landingTargetMeasure(y) == OK) {
+		mavlink_and_console_log_info(&mavlink_log_pub, "Landing target init");
+		_sensorTimeout &= ~SENSOR_TARGET;
+		_sensorFault &= ~SENSOR_TARGET;
+	}
+}
+
+int BlockLocalPositionEstimator::landingTargetMeasure(Vector<float, n_y_target> &y)
+{
+	if (_target_mode.get() == Target_Stationary) {
+		if (_sub_landing_target_pose.get().rel_vel_valid) {
+			y(0) = _sub_landing_target_pose.get().vx_rel;
+			y(1) = _sub_landing_target_pose.get().vy_rel;
+			_time_last_target = _timeStamp;
+
+		} else {
+			return -1;
+		}
+
+		return OK;
+
+	}
+
+	return -1;
+}
+
+void BlockLocalPositionEstimator::landingTargetCorrect()
+{
+	if (_target_mode.get() == Target_Moving) {
+		// nothing to do in this mode
+		return;
+	}
+
+	// measure
+	Vector<float, n_y_target> y;
+
+	if (landingTargetMeasure(y) != OK) { return; }
+
+	// calculate covariance
+	float cov_vx = _sub_landing_target_pose.get().cov_vx_rel;
+	float cov_vy = _sub_landing_target_pose.get().cov_vy_rel;
+
+	// use sensor value only if reasoanble
+	if (cov_vx < _target_min_cov.get() || cov_vy < _target_min_cov.get()) {
+		cov_vx = _target_min_cov.get();
+		cov_vy = _target_min_cov.get();
+	}
+
+	// target measurement matrix and noise matrix
+	Matrix<float, n_y_target, n_x> C;
+	C.setZero();
+	// residual = (y + vehicle velocity)
+	// sign change because target velocitiy is -vehicle velocity
+	C(Y_target_x, X_vx) = -1;
+	C(Y_target_y, X_vy) = -1;
+
+	// covariance matrix
+	SquareMatrix<float, n_y_target> R;
+	R.setZero();
+	R(0, 0) = cov_vx;
+	R(1, 1) = cov_vy;
+
+	// residual
+	Vector<float, n_y_target> r = y - C * _x;
+
+	// residual covariance, (inverse)
+	Matrix<float, n_y_target, n_y_target> S_I =
+		inv<float, n_y_target>(C * _P * C.transpose() + R);
+
+	// fault detection
+	float beta = (r.transpose()  * (S_I * r))(0, 0);
+
+	if (beta > BETA_TABLE[n_y_target]) {
```

Comment:
You need to generate the value for `n_y_target` on `BETA_TABLE`. Use `src/modules/local_position_estimator/fault_table.py` for that.